### PR TITLE
Add e2e tests for typical contracts #918 new test template

### PR DESCRIPTION
**Pull request by lzxm160**
work on #918 ,using new test template

### CODE REVIEW

**Comment by lzxm160** [comment]
add array-return.sol,basic-token.sol,call-dynamic.sol,f.value.sol,factory.sol,mapping-delete.sol,modifiers.sol

File: action/protocol/execution/protocol_test.go [hunk 1]
```diff
@@ -153,7 +154,7 @@ func (cfg *ExecutionConfig) ExpectedReturnValue() []byte {
 
 type SmartContractTest struct {
 	InitBalances []ExpectedBalance `json:"initBalances"`
-	Deploy       ExecutionConfig   `json:"deploy"`
+	Deploy       []ExecutionConfig `json:"deploy"`
```
Comment (CoderZhi) [inline action/protocol/execution/protocol_test.go]: Deploys

File: action/protocol/execution/protocol_test.go [hunk 2]
```diff
@@ -267,26 +268,29 @@ func (sct *SmartContractTest) deployContract(
...
+		if sct.Deploy[i].ExpectedGasConsumed() != 0 {
+			r.Equal(sct.Deploy[i].ExpectedGasConsumed(), receipt.GasConsumed)
+		}
 
+		ws, err := bc.GetFactory().NewWorkingSet()
+		r.NoError(err)
+		stateDB := evm.NewStateDBAdapter(bc, ws, uint64(0), hash.ZeroHash256)
+		var evmContractAddrHash common.Address
+		addr, _ := address.FromString(contractAddress)
+		fmt.Printf("contract address:0x%x\n", addr.Bytes())
```
Comment (CoderZhi) [inline action/protocol/execution/protocol_test.go]: delete

File: action/protocol/execution/protocol_test.go [hunk 3]
```diff
@@ -267,26 +268,29 @@ func (sct *SmartContractTest) deployContract(
...
-	}
-
-	ws, err := bc.GetFactory().NewWorkingSet()
-	r.NoError(err)
-	stateDB := evm.NewStateDBAdapter(bc, ws, uint64(0), hash.ZeroHash256)
-	var evmContractAddrHash common.Address
-	addr, _ := address.FromString(contractAddress)
-	copy(evmContractAddrHash[:], addr.Bytes())
-	r.True(bytes.Contains(sct.Deploy.ByteCode(), stateDB.GetCode(evmContractAddrHash)))
+	var contractAddress string
```
Comment (CoderZhi) [inline action/protocol/execution/protocol_test.go]: change to string[]

**Pull request merged**